### PR TITLE
Add Policy Tests

### LINKED ISSUE #734: Write tests for lib/code_corps/helpers/policy.ex
# Problem

This is a helper that we use often, but is completely untested at the moment. It's tested as a side effect of testing individual policies, but we should still test it directly.

## Subtasks
- [ ] Write tests for all public functions of lib/code_corps/helpers/policy.ex

As a good pointer, look at the `@spec` statements and `@doc` blocks for each public function. Tests should make sure all cases mentioned in those two are covered.

### PR DESCRIPTION
**Pull request by securingsincity**
# What's in this PR?
Adds tests for Policy Helpers

## References
Fixes #734 

Progress on: #

### CODE REVIEW

**Comment by securingsincity** [comment]
Still a WIP but getting the PR started

**Comment by securingsincity** [comment]
This should be all set.

**Comment by joshsmith** [comment]
@securingsincity thanks! I'll take a look at this tomorrow morning.

File: test/lib/code_corps/helpers/policy_test.exs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+  describe "owned_by/2" do
+    test "returns false when Organization owner_id doesn't match User user_id" do
+      refute Policy.owned_by?(%Organization{owner_id: 1}, %User{id: 2})
+    end
+
+    test "returns false when invalid arguments are passed" do
+      refute Policy.owned_by?(nil, 2)
+    end
+    test "returns false if a project is not owned by the user" do
```
Comment (begedin) [inline test/lib/code_corps/helpers/policy_test.exs]: Add a space here.

File: test/lib/code_corps/helpers/policy_test.exs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+  def create_project_user_with_role(role) do
+    user = insert(:user)
+    project = insert(:project)
+    insert(:project_user, project: project, user: user, role: role)
+    {project, user}
+  end
+
+  describe "owned_by/2" do
+    test "returns false when Organization owner_id doesn't match User user_id" do
```
Comment (begedin) [inline test/lib/code_corps/helpers/policy_test.exs]: I would reword this one and the other one for organization to 

`"returns false when organization is not owned by user"`
`"returns true when organization is owned by user"`

We want the tests to talk about "what we want the app to do" not "what's implemented".

File: test/lib/code_corps/helpers/policy_test.exs [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    test "should return task of a Changeset" do
+      task = insert(:task)
+      changeset = %Changeset{changes: %{task_id: task.id}}
+      result = Policy.get_task(changeset)
+      assert result.id == task.id
+    end
+  end
+
+  describe "task_authored_by?/1" do
+    test "should return true if the Task user_id and User id match" do
```
Comment (begedin) [inline test/lib/code_corps/helpers/policy_test.exs]: Again, probably word these as 

`"returns true if the user is the author of the task"`
`"returns false if the user is not the author of the task"`

**Review by begedin — CHANGES_REQUESTED** [review]
Sorry for the late review. There is just some minor wording we would like to change. The code itself looks great and will be ready to merge as soon as you're done.

**Review by begedin — APPROVED** [review]
Looks great. Rebase and merge when ready. 👍

**Comment by securingsincity** [comment]
@begedin should be squashed now. Thanks.
